scipy/linalg/tests/_cython_examples/extending.pyx

```diff
@@ -3,6 +3,7 @@
 #cython: boundscheck=False
 #cython: wraparound=False
 
+cimport scipy.linalg
```

Comment:
FWIW, both this and the original reproducer already pass for me locally on Linux with SciPy `main`:

```diff
--- a/scipy/linalg/tests/_cython_examples/extending.pyx
+++ b/scipy/linalg/tests/_cython_examples/extending.pyx
@@ -3,6 +3,8 @@
 #cython: boundscheck=False
 #cython: wraparound=False
 
+cimport scipy.linalg.cython_lapack
+cimport scipy.linalg
 from scipy.linalg.cython_blas cimport cdotu
 from scipy.linalg.cython_lapack cimport dgtsv
```

I don't know if we'll have much luck reproducing in the testsuite, since the reports in the issue seem a bit flaky perhaps.